extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -85,10 +85,18 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
       return;
     }
     this.mutateElement(() => {
+      let shopTagCSS = shoppingTagCSS;
+      if (document['dir'] === 'rtl') {
+        shopTagCSS = shoppingTagCSS.replace(
+          '.amp-story-shopping-tag-inner{',
+          '.amp-story-shopping-tag-inner{flex-flow:row-reverse !important;'
+        );
+      }
```

Comment:
This should be in the CSS file. It separates concerns and makes maintenance easier.
Please follow [this pattern](https://github.com/ampproject/amphtml/blob/main/extensions/amp-story/1.0/amp-story-system-layer.js#L749-L760) to set the rtl attribute on the shadow dom element.

Reply:
We cannot use this pattern as is as the state update is fired before the shadow root has been created.
This is for preserving the dot position to the left, not toggling the rtl stuff on the shadow root. Let's discuss a workaround.

Reply:
> as the state update is triggered before the shadow root has been created

We could get the state and set RTL before the element is appended.
like:
```
    this.mutateElement(() => {
      this.shoppingTagEl_ = renderShoppingTagTemplate(tagData);
      this.onRtlStateUpdate_(this.storeService_.get(StateProperty.RTL_STATE));
      createShadowRootWithStyle(
        this.element,
        this.shoppingTagEl_,
        shoppingTagCSS
      );
    });
  }
```

Then use a reference to `shoppingTagEl_` to set the dir attribute.